robottelo/cli/factory.py

```diff
@@ -3659,148 +3662,30 @@ def setup_capsule_virtual_machine(capsule_vm, org_id=None, lce_id=None,
-                .format(err.msg)
-            )
-    # Publish the content view
-    try:
-        ContentView.publish({u'id': content_view_id})
-    except CLIReturnCodeError as err:
-        raise CLIFactoryError(
-            u'Failed to publish new version of content view\n{0}'
-            .format(err.msg)
-        )
-    # Get the latest content view version id
-    try:
-        content_view_version = ContentView.info(
-            {u'id': content_view_id}
-        )['versions'][-1]
-    except CLIReturnCodeError as err:
-        raise CLIFactoryError(
-            u'Failed to fetch content view info\n{0}'.format(err.msg))
-    # Promote content view version to lifecycle environment
-    try:
-        ContentView.version_promote({
-            u'id': content_view_version['id'],
-            u'organization-id': org_id,
-            u'to-lifecycle-environment-id': lce_id,
-        })
-    except CLIReturnCodeError as err:
-        raise CLIFactoryError(
-            u'Failed to promote version to next environment\n{0}'
-            .format(err.msg)
-        )
-    activation_key = make_activation_key({
-        u'organization-id': org_id,
-        u'lifecycle-environment-id': lce_id,
-        u'content-view-id': content_view_id,
-        u'name': 'capsule-{0}'.format(gen_alphanumeric())
-    })
-    ActivationKey.update({
-        'auto-attach': 0,
-        'id': activation_key['id'],
-        'organization-id': org_id,
-    })
-    # Add subscriptions to activation-key
-    subscriptions = Subscription.list({'organization-id': org_id})
-    for subscription in subscriptions:
-        activationkey_add_subscription_to_repo({
-            'organization-id': org_id,
-            'activationkey-id': activation_key['id'],
-            'subscription': subscription['name']
-        })
+    content_setup_data = setup_cdn_and_custom_repos_content(
+        org_id=org_id,
+        lce_id=lce_id,
+        repos=rh_repos,
+        upload_manifest=True,
+        rh_subscriptions=[
+            DEFAULT_SUBSCRIPTION_NAME,
+            SATELLITE_SUBSCRIPTION_NAME,
+        ]
+    )
     # Install katello ca on capsule virtual machine
     capsule_vm.install_katello_ca()
     # Register the capsule host to satellite
     capsule_vm.register_contenthost(
         capsule_org['name'],
-        activation_key=activation_key['name']
+        activation_key=content_setup_data['activation_key']['name']
     )
     # Patch the os release version
-    capsule_vm.run(
-        "touch /etc/yum/vars/releasever "
-        "&& echo '{0}' > /etc/yum/vars/releasever"
-        .format(rh_product_releasever)
-    )
+    capsule_vm.patch_os_release_version(distro=capsule_vm.capsule_distro)
+
     # Enable the repositories
     for repo in rh_repos:
-        repo_id = repo['repository-id']
-        capsule_vm.enable_repo(repo_id)
+        if repo['cdn']:
+            capsule_vm.enable_repo(repo['repository-id'])
```

Comment:
If it's cdn repo for sure - we'd better force enabling with `force=True` not to use default enabling logic which checks robottelo.properties and may introduce failures later (like we have right now with BZ1526766).